### ISSUE TITLE
Some ArcGIS Portal Vector Tiles do not work

### ISSUE THREAD

**Issue by ca0v** [openlayers/ol-mapbox-style#344]
In some cases the VectorTileServer response is HTML instead of JSON.

When a VectorTileServer has a style with a relative "url" path it is computing the path to the VectorTileServer like this:

`url = withPath(url, path);` and `url = host + url`

In this case, that would be,
`url = "/server/rest/services/Hosted/Folder/VectorTileServer/resources/styles/../../"`
`path = "/server/rest/services/Hosted/England/VectorTileServer/resources/styles/"`

The final url looks like https://some.machine/server/rest/services/Hosted/Folder/VectorTileServer/resources/styles/../../

Then `setupVectorLayer` creates TileJSON, which queries the url and **gets HTML instead of JSON**.
The layer fails to render.

If I set a breakpoint here and modify the options.url to be https://some.machine/server/rest/services/Hosted/Folder/VectorTileServer?f=json, then it works:

`client.open('GET', options.url)`

So the fix would be to have a way to forward query parameter or otherwise modify the url:

`url = new URL(host + url).href;`
`url += "?f=json";`

Here is the actual style from https://some.machine/server/rest/services/Hosted/Folder/VectorTileServer/resources/styles/root.json

`{
  "version": 8,
  "sprite": "../sprites/sprite",
  "glyphs": "../fonts/{fontstack}/{range}.pbf",
  "sources": {
    "esri": {
      "type": "vector",
      "bounds": [
        -6.96817,
        48.7215,
        5.58361,
        58.92
      ],
      "minzoom": 0,
      "maxzoom": 19,
      "scheme": "xyz",
      "url": "../../"
    }
  },
  "layers": [
    {
      "id": "buildings",
      "type": "fill",
      "source": "esri",
      "source-layer": "buildings",
      "layout": {},
      "paint": {
        "fill-color": "#FCC2CE",
        "fill-outline-color": "#6E6E6E"
      }
    },
    {
      "id": "places",
      "type": "circle",
      "source": "esri",
      "source-layer": "places",
      "layout": {},
      "paint": {
        "circle-radius": 2.2,
        "circle-color": "#A13B53",
        "circle-stroke-color": "#000000",
        "circle-stroke-width": 0.933333
      }
    }
  ]
}`

**Comment by ahocevar** [comment]
So the problem is the missing `?f=json` which is not carried along? Then we should probably add the `Accept: application/json` header to the request. Could you try if that would work in your case? Using curl or Postman, or something similar?

**Comment by ca0v** [comment]
It did not work.  Interestingly https://basemaps.arcgis.com/arcgis/rest/services/World_Basemap_v2/VectorTileServer returns JSON but when I publish a similar service with a similar URL it returns HTML.  The Accept: application/json had no affect on the response.  I cannot explain why they are not both HTML or both JSON.

![image](https://user-images.githubusercontent.com/1585764/133136810-21227eb6-6708-47c1-a9cd-e191a451e8ad.png)

I see no way to solve this apart from introducing an options parameter to olms or an entirely new function.  

Please take these suggestions with a grain of salt...

`await olms({map, tile:url1, style: url2}); // separates the style information from the actual vector service`

or

```
l = await createLayer({ tile:url1}); // or create an unstyled layer
await styleLayer(l, {style: url}); // and then style that layer
```

**Comment by ahocevar** [comment]
Could it be that these requests go through a proxy which does not pass the headers along?

**Comment by ca0v** [comment]
They go through the arcgis web adapter but I can't speak to what gets forwarded or not. There is no 3rd party proxy. I will try again on port 6443 to bypass the adaptor.

**Comment by larssoe74** [comment]
I am experiencing the same problem:

It seems as ol-mapbox-style for ArcGIS Vector Tile currently only works with https://basemaps.arcgis.com - not self-hosted ArcGIS Server or ArcGIS Portal.
[Stackoverflow](https://stackoverflow.com/questions/69917241/openlayers-arcgis-vector-tiles-styling-with-ol-mapbox-style)

**Comment by kasun-linz** [comment]
Hi,
Is there any solution for this one yet?

I tried to fetch the json and pass it to the map to eliminate issue caused by missing query parameter 'f=json'. However, I am still see the same issue even though I see JSON is correctly passing to the function.

However same code works when I change the baseMapUrl to one of the Esri global ones. I can't see much difference in layers section in the json two URLs produce.

Esri basemap url
`https://basemaps-api.arcgis.com/arcgis/rest/services/styles/ArcGIS:Streets?type=style&token=<token>`

Error I see in the console.
![image](https://user-images.githubusercontent.com/74882764/148703438-5b98513c-0d86-42a0-a704-6922280c1353.png)


Thanks,
Kasun

```
    const baseMapStyleUrl = 'https://tiles.arcgis.com/tiles/XTtANUDT8Va4DLwI/arcgis/rest/services/nz_vector_basemap_v1/VectorTileServer/resources/styles?f=json'
    
    fetch(baseMapStyleUrl)
      .then(res => res.json())
      .then(styleJson => {
        console.log(styleJson);
        return olms(map, styleJson)
      })
      .then(map => map.addLayer(vectorTileLayer));
```

**Comment by ahocevar** [comment]
There is another issue with this style. It Esri's use of relative URLs, which ol-mapbox-style does not handle. However, OpenLayers's `MapboxVector` layer handles those. But with the `MapboxVector` layer, the TileJSON cannot be fetched either, because of a missing token.

I think the only way to fix this in a user-friendly way would be to introduce something like mapbox-gl's [`transformRequest`](https://github.com/mapbox/mapbox-gl-js/pull/9225#issuecomment-578089885). Then you could make paths absolute and add tokens for every url that needs to be requested (TileJSON, sprites, tiles).

**Comment by ahocevar** [comment]
`transformRequest` implemented in v8.0.0.
